api/app/signals/apps/api/v1/private/views.py

```diff
@@ -160,49 +157,41 @@ class SignalCategoryRemovedAfterViewSet(viewsets.GenericViewSet, mixins.ListMode
-                                  mixins.DestroyModelMixin, mixins.UpdateModelMixin):
-    serializer_class = StatusMessageTemplateSerializer
-    serializer_detail_class = StatusMessageTemplateSerializer
+class StatusMessageTemplatesViewSet(mixins.RetrieveModelMixin, mixins.CreateModelMixin,
+                                    viewsets.GenericViewSet):
+    serializer_class = StateStatusMessageTemplateSerializer
 
     authentication_classes = (JWTAuthBackend,)
     permission_classes = (SIAPermissions & ModelWritePermissions,)
     pagination_class = None
 
-    queryset = StatusMessageTemplate.objects.all()
+    queryset = StatusMessageTemplate.objects.none()
 
-    def create(self, request, *args, **kwargs):
-        many = type(request.data) in [list, tuple]
-
-        serializer = self.get_serializer(data=request.data, many=many)
-        serializer.is_valid(raise_exception=True)
-
-        try:
-            self.perform_create(serializer)
-        except CoreValidationError as e:
-            raise ValidationError(e.message)
-
-        headers = self.get_success_headers(serializer.data)
-        return Response(serializer.data, status=HTTP_201_CREATED, headers=headers)
+    def get_object(self):
+        if 'slug' in self.kwargs and 'sub_slug' in self.kwargs:
+            kwargs = {'parent__slug': self.kwargs['slug'],
+                      'slug': self.kwargs['sub_slug']}
+        else:
+            kwargs = {'slug': self.kwargs['slug']}
 
-    def update(self, request, *args, **kwargs):
-        partial = kwargs.pop('partial', False)
-        instances = []
-        for data in self.request.data:
-            self.kwargs['pk'] = data['pk']
-            instance = self.get_object()
+        obj = get_object_or_404(Category.objects.all(), **kwargs)
+        self.check_object_permissions(self.request, obj)
+        return obj
 
-            serializer = self.get_serializer(instance, data=data, partial=partial, many=False)
-            serializer.is_valid(raise_exception=True)
-            self.perform_update(serializer)
+    def get_serializer_context(self):
+        context = super(StatusMessageTemplatesViewSet, self).get_serializer_context()
+        context.update({'category': self.get_object()})
+        return context
 
-            if getattr(instance, '_prefetched_objects_cache', None):
-                instance._prefetched_objects_cache = {}
-            instances.append(instance)
+    def retrieve(self, request, *args, **kwargs):
+        status_message_templates = self.get_object().status_message_templates.all()
+        serializer = self.get_serializer(status_message_templates, many=True)
+        return Response(serializer.data)
 
-        serializer = self.get_serializer(instances, many=True)
-        return Response(serializer.data, status=HTTP_200_OK)
+    def create(self, request, *args, **kwargs):
+        many = isinstance(request.data, list)
```

Comment:
Isn't our data always in an array? Like:
```
[{"state": "o", "templates": [{"title": "een titel", "text": "een tekst"}, ...]}, ...]
```
If so, should we HTTP 400 when receiving something not wrapped in an array?